### PR TITLE
Add command to migrate legacy pickled Result objects to datastore

### LINKED ISSUE #947: DB Migration script
## Acceptance Criteria 

- Create DB migration script for users with legacy dispatch db to migrate to post result object decomposition DB

### PR DESCRIPTION
**Pull request by cjao**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Closes #947, at least for pickle files created with v0.110.2 (the latest released version on PyPI).

### CODE REVIEW

**Review by scottwn — APPROVED** [review]

File: doc/source/api/api.rst [hunk 1]
```diff
@@ -200,5 +200,5 @@ This Command Line Interface (CLI) tool is used to manage Covalent server.
 
 .. click:: covalent_dispatcher._cli.cli:cli
     :prog: covalent
-    :commands: start,stop,restart,status,purge,logs
+    :commands: start,stop,restart,status,purge,logs,migrate_legacy_result_object
```
Comment (santoshkumarradha) [inline doc/source/api/api.rst:203]: @cjao for some reason I dont see it here https://covalent--976.org.readthedocs.build/en/976/api/api.html#covalent-cli-tool

Comment (cjao) [reply doc/source/api/api.rst:203]: I don't see it either. Thanks for the catch!

Comment (cjao) [reply doc/source/api/api.rst]: Forgot that `click` converts underscores to hyphens.

**Review by scottwn — APPROVED** [review]

**Review by santoshkumarradha — APPROVED** [review]
Awesome good to go from docs side !

**Review by scottwn — APPROVED** [review]

**Review by santoshkumarradha — APPROVED** [review]
GTG 🚀

**Pull request merged**